node/src/main/scala/coop/rchain/node/web/https4s/RouterFix.scala

```diff
@@ -0,0 +1,61 @@
+package coop.rchain.node.web.https4s
+
+import cats.data.Kleisli
+import cats.syntax.all._
+import cats.{Functor, Monad}
+import org.http4s.server.Router
+import org.http4s.{HttpRoutes, Request}
+
+/**
+  * Original code from http4s [[Router]] with the fix for mappings to work with endpoints4s Path decoder.
+  * The difference is in the [[RouterFix.translate]] function.
+  *
+  * https4s [[Router]] uses [[Request.attributes]] to set uri caret position
+  *  which is not supported by _endpoints4s_ Path decoder.
+  * This fix changes [[Request.uri]] instead by removing path prefix defined
+  *  in mapping.
+  */
+object RouterFix {
+
+  /**
+    * Defines an [[HttpRoutes]] based on list of mappings.
+    */
+  def apply[F[_]: Monad](mappings: (String, HttpRoutes[F])*): HttpRoutes[F] =
+    define(mappings: _*)(HttpRoutes.empty[F])
+
+  /**
+    * Defines an [[HttpRoutes]] based on list of mappings and
+    * a default Service to be used when none in the list match incoming requests.
+    *
+    * The mappings are processed in descending order (longest first) of prefix length.
+    */
+  def define[F[_]: Monad](
+      mappings: (String, HttpRoutes[F])*
+  )(default: HttpRoutes[F]): HttpRoutes[F] =
+    mappings.sortBy(_._1.length).foldLeft(default) {
```

Comment:
Just curious. Why sort here?

Reply:
This is direct copy from http4s source.
https://github.com/http4s/http4s/blob/519b72f3/server/shared/src/main/scala/org/http4s/server/Router.scala#L40

Mapping are sorted to match the longest first e.g. if mappings are defined as `/api` and `/api/v1`, request to `/api/v1/blocks` should not match the first (short) segment `/api` and fail to find rest of the route in this mapping. In other words mappings are like sub-roots so we want to find the longest that can be matched and then go deeper.